repose-aggregator/extensions/simple-rbac/src/main/scala/org/openrepose/filters/simplerbac/SimpleRbacFilter.scala

```diff
@@ -0,0 +1,261 @@
+      this,
+      classOf[SimpleRbacConfig]
+    )
+  }
+
+  override def destroy(): Unit = {
+    configurationService.unsubscribeFrom(configurationFile, this)
+  }
+
+  override def doFilter(servletRequest: ServletRequest, servletResponse: ServletResponse, filterChain: FilterChain): Unit = {
+    if (!initialized) {
+      logger.error("Simple RBAC filter has not yet initialized...")
+      servletResponse.asInstanceOf[HttpServletResponse].sendError(500)
+    } else {
+      val mutableHttpRequest = MutableHttpServletRequest.wrap(servletRequest.asInstanceOf[HttpServletRequest])
+      val mutableHttpResponse = MutableHttpServletResponse.wrap(mutableHttpRequest, servletResponse.asInstanceOf[HttpServletResponse])
+
+      logger.trace("Simple RBAC filter processing request...")
+      validator.validate(mutableHttpRequest,mutableHttpResponse,filterChain)
+    }
+    logger.trace("Simple RBAC filter returning response...")
+  }
+
+  override def configurationUpdated(configurationObject: SimpleRbacConfig): Unit = {
+    configuration = configurationObject
+    config.enableRaxRolesExtension = true
+    config.checkPlainParams = true
+    config.maskRaxRoles403 = configuration.isEnableMasking403S
+    config.setResultHandler(getHandlers)
+
+    val rbacWadl = rbacToWadl(Option(configuration.getResources)).orElse(
+      Option(configuration.getResourcesFileName: String) match {
+        case Some(n) =>
+          rbacToWadl(readResource(
+              configurationService.getResourceResolver.resolve(n).newInputStream()
+          ))
+        case _ =>
+          None
+      }
+    )
+    rbacWadl match {
+      case Some(r) =>
+        Option(configuration.getWadlOutput: String) match {
+          case Some(wadlOutput) =>
+            val wadlPath: Path = Paths.get(StringUriUtilities.formatUri(getPath(wadlOutput, configurationRoot))).toAbsolutePath
+            try {
+              if (Files.exists(wadlPath) && Files.isWritable(wadlPath) || !Files.exists(wadlPath) && Files.createFile(wadlPath) != null) {
+                Files.write(wadlPath, r.getBytes)
+                logger.debug("Saved WADL to: " + wadlPath)
+              } else {
+                logger.warn("Cannot write to WADL file: " + wadlPath)
+                logger.debug(s"Generated WADL:\n\n$r\n")
+              }
+            } catch {
+              case ex: IOException =>
+                logger.warn("Cannot write to WADL file: " + wadlPath, ex)
+                logger.debug(s"Generated WADL:\n\n$r\n")
+            }
+          case _ =>
+            logger.debug(s"Generated WADL:\n\n$r\n")
+        }
+        val uuid = UUID.randomUUID
+        validator = Validator.apply(
+          s"SimpleRbacValidator_$uuid",
+          new StreamSource(new ByteArrayInputStream(r.getBytes), "file://simple-rbac.wadl"),
+          config
+        )
+        initialized = true
+      case _ =>
+        logger.warn("Unable to generate the WADL; check the provided resources.")
+    }
+  }
```

Comment:
i didn't see you tear down the old validator. IS that something you need to do? I know we have issues around that ni the validator filter.